android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -140,23 +163,100 @@ private void updateState() {
     }
 
     private void updateStatus() {
-        assert mBraveWalletService != null;
         AccountsPermissionsHelper accountsPermissionsHelper = new AccountsPermissionsHelper(
-                mBraveWalletService, mAccountInfos, getCurrentHostHttpAddress());
+                mBraveWalletPanelServices.getBraveWalletService(), mAccountInfos, getCurrentHostHttpAddress());
         accountsPermissionsHelper.checkAccounts(() -> {
-            HashSet<AccountInfo> mAccountsWithPermissions =
+            mAccountsWithPermissions =
                     accountsPermissionsHelper.getAccountsWithPermissions();
-            if (mAccountsWithPermissions.size() != 0) {
-                mBtnConnectedStatus.setText(mPopupView.getResources().getString(
-                        R.string.dapp_wallet_panel_connectivity_status_connected));
-                mBtnConnectedStatus.setCompoundDrawablesRelativeWithIntrinsicBounds(
-                        R.drawable.ic_check_white, 0, 0, 0);
+            updateAccount();
+        });
+    }
+
+    private void updateAccount() {
+        mBraveWalletPanelServices.getKeyringService().getSelectedAccount(CoinType.ETH, address -> {
+            String selectedAccount = "";
+            if (address != null && !address.isEmpty()) {
+                selectedAccount = address;
+                updateAccountInfo(selectedAccount);
+            } else {
+                if (mAccountsWithPermissions.size() != 0) {
+                    selectedAccount = mAccountsWithPermissions.iterator().next().address;
+                } else if (mAccountInfos.length != 0) {
```

Comment:
nit: can replace `!= 0` with `> 0`  or `!mAccountsWithPermissions.isEmpty()`

Reply:
not much, it just quickly makes it clear that it should be greater than one, as `!=0` can be positive or negative.